librepo/yum.c

```diff
@@ -131,8 +131,18 @@ lr_yum_repo_update(LrYumRepo *repo, const char *type, const char *path)
 
 /* main bussines logic */
 
+gint
+compare_records(gconstpointer a, gconstpointer b)
+{
+    GSList* elem = (GSList*) a;
+    LrYumRepoMdRecord* yum_record = (LrYumRepoMdRecord*) elem;
+    char *type1 = (char *) yum_record->type;
+    char *type2 = (char *) b;
+    return strcmp(type1, type2);
```

Comment:
Maybe use g_strcmp0() as it's more bullet-prof (NULL won't cause SIGSEGV)

https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strcmp0